File: Python/Two Pointer/11. Container With Most Water.py
""" 11. Container With Most Water
Difficulty: Medium
Submission Link: https://leetcode.com/problems/container-with-most-water/submissions/1247602575/ """

class Solution(object):
    def maxArea(self, height):
        ans=0
        s=0
        e=len(height)-1
        while s<e:
            area=(e-s)*min(height[s],height[e])
            ans=max(area,ans)
            if height[s]<height[e]:
                s+=1
            else:
                e-=1
        return ans

File: C/Two Pointer/11. Container With Most Water.c
/* 11. Container With Most Water
Difficulty: Medium
Submission Link: https://leetcode.com/problems/container-with-most-water/submissions/1247603052/
 */

int maxArea(int* height, int heightSize) {
    int left = 0;
    int right = heightSize - 1;
    int maxWater = 0;
    while (left < right) {
        int width = right - left;
        int h = height[left] < height[right] ? height[left] : height[right];
        int water = width * h;
        if (water > maxWater) {
            maxWater = water;
        }if (height[left] < height[right]) {
            left++;
        } else {
            right--;
        }
    }

    return maxWater;
}